python/lsst/pipe/tasks/fit_coadd_multiband.py

```diff
@@ -0,0 +1,321 @@
+        bands_read_only : `set`
+            The set of bands that the subtask will only read data
+            (measurement catalog and exposure) for.
+        """
+        try:
+            bands_fit = self.fit_coadd_multiband.bands_fit
+        except AttributeError:
+            raise RuntimeError(f'{__class__}.fit_coadd_multiband must have bands_fit attribute') from None
+        bands_read_only = self.fit_coadd_multiband.bands_read_only()
+        return set(bands_fit), set(bands_read_only)
+
+
+class CoaddMultibandFitTask(pipeBase.PipelineTask):
+    """Fit deblended exposures in multiple bands simultaneously.
+
+    It is generally assumed but not enforced (except optionally by the
+    configurable `fit_coadd_multiband` subtask) that there is only one exposure
+    per band, presumably a coadd.
+    """
+    ConfigClass = CoaddMultibandFitConfig
+    _DefaultName = "CoaddMultibandFit"
+
+    def __init__(self, initInputs, **kwargs):
+        super().__init__(initInputs=initInputs, **kwargs)
+        self.makeSubtask("fit_coadd_multiband")
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        id_tp = ExposureIdInfo.fromDataId(butlerQC.quantum.dataId, "tract_patch").expId
+        # This is a roundabout way of ensuring all inputs get sorted and matched
+        input_refs_objs = [(getattr(inputRefs, key), inputs[key])
+                           for key in ("cats_meas", "coadds", "psf_fits")]
+        cats, exps, psf_fits = [
+            {dRef.dataId: obj for dRef, obj in zip(refs, objs)}
+            for refs, objs in input_refs_objs
+        ]
+        dataIds = set(cats).union(set(exps))
+        catexps = [
+            CatalogExposureInputs(
+                catalog=cats[dataId], exposure=exps[dataId], table_psf_fits=psf_fits[dataId],
+                dataId=dataId, id_tract_patch=id_tp,
+            )
+            for dataId in dataIds
+        ]
+        outputs = self.run(catexps=catexps, cat_ref=inputs['cat_ref'])
+        butlerQC.put(outputs, outputRefs)
+
+    def run(self, catexps: List[CatalogExposure], cat_ref: afwTable.SourceCatalog) -> pipeBase.Struct:
+        """Fit sources from a reference catalog using data from multiple
+        exposures in the same region (patch).
+
+        Parameters
+        ----------
+        catexps : `typing.List [CatalogExposure]`
+            A list of catalog-exposure pairs in a given band.
+        cat_ref : `lsst.afw.table.SourceCatalog`
+            A reference source catalog to fit.
+
+        Returns
+        -------
+        retStruct : `lsst.pipe.base.Struct`
+            A struct with a cat_output attribute containing the output
+            measurement catalog.
+
+        Notes
+        -----
+        Subtasks may have further requirements; see `CoaddMultibandFitSubTask.run`.
+        """
+        cat_output = self.fit_coadd_multiband.run(catalog_multi=cat_ref, catexps=catexps).output
```

Comment:
You'll need code here to add the footprints from the scarlet models, otherwise your individual sources won't have footprints.